core/DataAccess/Model.php

```diff
@@ -238,13 +248,37 @@ public function updateArchiveAsInvalidated($archiveTable, $idSites, $allPeriodsT
             }
         }
 
-        $fields = ['idarchive', 'name', 'report', 'idsite', 'date1', 'date2', 'period', 'ts_invalidated'];
-
-        Db\BatchInsert::tableInsertBatch(Common::prefixTable('archive_invalidations'), $fields, $dummyArchives);
+        if (!empty($dummyArchives)) {
+            $fields = ['idarchive', 'name', 'report', 'idsite', 'date1', 'date2', 'period', 'ts_invalidated'];
+            Db\BatchInsert::tableInsertBatch(Common::prefixTable('archive_invalidations'), $fields, $dummyArchives);
+        }
 
         return count($idArchives);
     }
 
+    private function getExistingInvalidations($idSites, $periodCondition, $nameCondition)
+    {
+        $table = Common::prefixTable('archive_invalidations');
+
+        $sql = "SELECT idsite, date1, date2, period, name, COUNT(*) as `count` FROM `$table`
+                 WHERE idsite IN (" . implode(',', $idSites) . ") AND status = " . ArchiveInvalidator::INVALIDATION_STATUS_QUEUED . "
```

Comment:
just seeing $idSites here. They are most likely already casted to int but be still great to add it again just to be safe and `array_map(intval, $idsites)`. I think further up around line 130 it's also the same

Reply:
👍